src/ConnectedKubernetes/ConnectedKubernetes.Autorest/custom/Set-AzConnectedKubernetes.ps1

```diff
@@ -0,0 +1,726 @@
+        if ($EnableGateway) {
+            if ($null -eq $GatewayResourceId) {
+                Write-Error "GatewayResourceId is required when EnableGateway is set."
+                return
+            }
+            $Null = $PSBoundParameters.Remove('EnableGateway')
+            $PSBoundParameters.Add('GatewayEnabled', $true)
+        }
+        # If DisableGateway is provided then set the gateway as disabled and remove gateway resourceId from parameters
+        if ($DisableGateway) {
+            $Null = $PSBoundParameters.Remove('DisableGateway')
+            $Null = $PSBoundParameters.Remove('GatewayResourceId')
+            $PSBoundParameters.Add('GatewayEnabled', $false)
+        }
+        
+        $CommonPSBoundParameters = @{}
+        if ($PSBoundParameters:HttpPipelineAppend) {
+            $CommonPSBoundParameters['HttpPipelineAppend'] = $HttpPipelineAppend
+        }
+        if ($PSBoundParameters:HttpPipelinePrepend) {
+            $CommonPSBoundParameters['HttpPipelinePrepend'] = $HttpPipelinePrepend
+        }
+        if ($PSBoundParameters:SubscriptionId) {
+            $CommonPSBoundParameters['SubscriptionId'] = $SubscriptionId
+        }
+        if ($PSBoundParametersPrivateLinkState -and ($null -ne $CustomLocationsOid) -and ($CustomLocationsOid -ne '')) {
+            Write-Warning "The features 'cluster-connect' and 'custom-locations' cannot be enabled for a private link enabled connected cluster."
+            $CustomLocationsOid = $null
+        }
+        if ($PSBoundParametersCustomLocationsOid) {
+            $Null = $PSBoundParameters.Remove('CustomLocationsOid')
+        }
+        $IdentityType = [Microsoft.Azure.PowerShell.Cmdlets.ConnectedKubernetes.Support.ResourceIdentityType]::SystemAssigned
+        $PSBoundParameters.Add('IdentityType', $IdentityType)
+        
+        # Region Deal with settings and protected settings
+        # If a new Kubernetes feature is added then code may need to be added here
+        # to suport protected settings as the Config DP is unable to process these
+        # themselves.  Add a check here to see if there are any that we currently
+        # do not suport.
+        $supportedFeatures = @("proxy")
+        if ($arcAgentrySettings) {
+            foreach ($key in $arcAgentrySettings.Keys) {
+                if (-not $supportedFeatures.Contains($key.ToLower())) {
+                    Write-Warning "Arc Agentry feature '${key}' is not supported for Connected Kubernetes"
+                }
+            }
+        }
+        if ($arcAgentryProtectedSettings) {
+            foreach ($key in $arcAgentryProtectedSettings.Keys) {
+                if (-not $supportedFeatures.Contains($key.ToLower())) {
+                    Write-Warning "ArcAgentry feature '${key}' is not supported for Connected Kubernetes"
+                }
+            }
+        }
+        #Endregion
+
+        # XW TODO: helm install can be extracted as common function to be used by both Set and New
```

Comment:
There are some `helm` helpers already so we can certainly put that there.  Sorry I did not spot this at the time.